### PR TITLE
feat(application-generic): Add the ability to have query parameters in bridge checks

### PR DESCRIPTION
**Pull request by elmdecoste**
### What changed? Why was the change needed?
We added the ability to have query parameters within the bridge urls. This is a requirement for any hosting service for the bridge where there are required query parameters for authentication. 

https://....url.../api/novu?x-vercel-protection-bypass=BYPASS_KEY

An example being Vercel's preview environments needing a query parameter for their bypass: https://vercel.com/docs/security/deployment-protection/methods-to-bypass-deployment-protection/protection-bypass-automation

### CODE REVIEW

File: apps/api/src/app/bridge/usecases/get-bridge-status/get-bridge-status.usecase.ts [hunk 1]
```diff
@@ -9,7 +9,10 @@ const axiosInstance = axios.create();
 export class GetBridgeStatus {
   async execute(command: GetBridgeStatusCommand): Promise<HealthCheck> {
     try {
-      const response = await axiosInstance.get<HealthCheck>(`${command.bridgeUrl}?action=health-check`, {
+      const bridgeActionUrl = new URL(command.bridgeUrl);
+      bridgeActionUrl.searchParams.set('action', 'health-check');
```
Comment (rifont) [inline apps/api/src/app/bridge/usecases/get-bridge-status/get-bridge-status.usecase.ts]: Thanks for refactoring this to use the `URL` API. We can additionally remove the hardcoded strings here with the following imports, creating a more intentional coupling between the Framework and API.

```typescript
import { PostActionEnum, GetActionEnum, HttpQueryKeysEnum } from '@novu/framework';
```

usage to replace (here and `preview-step.usecase`):
`'action'` -> `PostActionEnum.ACTION`
`'health-check'` -> `GetActionEnum.HEALTH_CHECK`
`'preview'` -> `PostActionEnum.PREVIEW`
`'workflowId'` -> `HttpQueryKeysEnum.WORKFLOW_ID`
`'stepId'` -> `HttpQueryKeysEnum.STEP_ID`

**Review by rifont — CHANGES_REQUESTED** [review]
Thank you for this improvement 🙏

Could you please replace the hardcoded strings with imports from `@novu/framework`, we'll release this improvement this week.

File: apps/api/src/app/bridge/usecases/get-bridge-status/get-bridge-status.usecase.ts (hunk 1 above)
Comment (elmdecoste) [reply apps/api/src/app/bridge/usecases/get-bridge-status/get-bridge-status.usecase.ts]: Fixed!

**Review by rifont — APPROVED** [review]
Thanks for the enum update! We will get this deployed tomorrow.

**Comment by rifont** [comment]
Hi @elmdecoste , this is now released to Novu Cloud 🙏

**Pull request merged**